core-common/src/test/java/org/zalando/nakadi/repository/kafka/KafkaRecordDeserializerTest.java

```diff
@@ -52,17 +62,155 @@ public void testDeserializeAvro() throws IOException {
                 data1
         );
 
-        Assert.assertEquals(
-                getExpectedNode0(),
-                new ObjectMapper()
-                        .readValue(deserializedEvent0, ObjectNode.class));
+        Assert.assertTrue(
+                getExpectedNode0(null).similar(new JSONObject(new String(deserializedEvent0))));
 
-        Assert.assertEquals(
-                getExpectedNode1(),
-                new ObjectMapper()
-                        .readValue(deserializedEvent1, ObjectNode.class));
+        Assert.assertTrue(
+                getExpectedNode1().similar(new JSONObject(new String(deserializedEvent1))));
     }
 
+    @Test
+    public void testDeserializeAvroMetadata0() throws IOException {
+        final var metadataVersion = "0";
+
+        final JSONObject jsonObject = new JSONObject()
+        .put("flow_id", "hek")
+        .put("partition", 0)
+        .put("received_at", SOME_TIME)
+        .put("published_by", "nakadi-test");
+
+        final var actualJson = getSerializedJsonObject(metadataVersion, jsonObject);
+        final var expectedJson = getExpectedNode0(jsonObject.put("received_at", SOME_TIME_DATE_STRING));
+        Assert.assertTrue(expectedJson.similar(actualJson));
```

Comment:
Is `similar` a strong enough test here?  Shouldn't we test for real equality?

Reply:
JSONObject does not ovveridde equals method, so it just checks referential equality. From docs about similar,
```
    /**
     * Determine if two JSONObjects are similar.
     * They must contain the same set of names which must be associated with
     * similar values.
     *
     * @param other The other JSONObject
     * @return true if they are equal
     */

```

Reply:
I found the docs as well.  But what does "similar values" really mean?  Is it a recursive application of the same method?  What about non-object values?  Is it enough for our test (e.g. do we ignore the actual values)?

Reply:
Yes, its a recursive function as long as its a `JSONObject` or `JSONArray`, but when we get to actual values (base case), the `equals()` method of the value is called. For non - object values (primitives?), the equals method of Primitive Wrapper classes should be getting called. Even though its called similar, it is the strongest equality check offered, perhaps it should just be called equals, although there be some reason they decided to call it similar instead (maybe order of elements is not important in similar, thats what I understand reading the implementation code)

Reply:
OK, makes sense.  Thank you :)